File: android/src/main/kotlin/com/tradechart/plugin/engine/ChartFrame.kt
package com.tradechart.plugin.engine

import android.graphics.Canvas
import android.graphics.RectF
import com.tradechart.plugin.bridge.generated.ConfigMessage
import com.tradechart.plugin.data.CandleStore
import com.tradechart.plugin.data.MarkerStore
import com.tradechart.plugin.theme.NativeChartTheme
import com.tradechart.plugin.viewport.ViewportCalculator

data class CrosshairState(
    val candleIndex: Int,
    val snappedX: Float,
    val y: Float,
)

data class ChartFrame(
    val canvas: Canvas,
    val candleStore: CandleStore,
    val markerStore: MarkerStore,
    val viewport: ViewportCalculator,
    val theme: NativeChartTheme,
    val config: ConfigMessage,
    val chartType: String,
    val width: Int,
    val height: Int,
    val contentRect: RectF,
    val mainChartRect: RectF,
    val volumeRect: RectF?,
    val priceAxisRect: RectF?,
    val timeAxisRect: RectF?,
    val crosshair: CrosshairState?,
)

File: ios/Classes/Engine/ChartFrame.swift
import CoreGraphics
import Foundation

struct CrosshairState {
  let candleIndex: Int
  let snappedX: CGFloat
  let y: CGFloat
}

struct ChartFrame {
  let context: CGContext
  let candleStore: CandleStore
  let markerStore: MarkerStore
  let viewport: ViewportCalculator
  let theme: NativeChartTheme
  let config: ConfigMessage
  let chartType: String
  let width: Int
  let height: Int
  let contentRect: CGRect
  let mainChartRect: CGRect
  let volumeRect: CGRect?
  let priceAxisRect: CGRect?
  let timeAxisRect: CGRect?
  let crosshair: CrosshairState?
}
